### PR TITLE
fix(datepicker): make datepicker work in compatibility mode

### LINKED ISSUE #4619: Datepicker doesn't work with NoConflictStyleCompatibilityMode
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Use `mat-datepicker` without any problem (with `NoConflictStyleCompatibilityMode`) inside angularjs-angular hybrid apps.

#### What is the current behavior?
It now throws error in the console (right after I click on the calendar icon to open datepicker):
> ERROR Error: The "md-" prefix cannot be used in ng-material v1 compatibility mode. It was used on an "button" element.

#### What are the steps to reproduce?
I created a github repo for demoing the issue: https://github.com/merobal/datepicker-demo
Demo page: https://merobal.github.io/datepicker-demo/

#### What is the use-case or motivation for changing an existing behavior?
We would like to use datepicker inside hybrid apps which also uses material1.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Latest angular, latest material, Win10, TS 2.2, all browsers.

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #4619

### CODE REVIEW

File: src/lib/datepicker/calendar.html [hunk 1]
```diff
@@ -1,16 +1,38 @@
 <div class="mat-calendar-header">
   <div class="mat-calendar-controls">
-    <button md-button class="mat-calendar-period-button" (click)="_currentPeriodClicked()"
-            [attr.aria-label]="_periodButtonLabel">
+
```
Comment (jelbourn) [inline src/lib/datepicker/calendar.html]: Add a TODO like
```html
<!-- TODO: stop doing this horrible hack for compatibility mode -->
```
?

File: src/lib/datepicker/calendar.ts [hunk 2]
```diff
@@ -111,6 +112,7 @@ export class MdCalendar<D> implements AfterContentInit {
   constructor(private _elementRef: ElementRef,
               private _intl: MdDatepickerIntl,
               private _ngZone: NgZone,
+              @Optional() @Inject(MATERIAL_COMPATIBILITY_MODE) public _isCompatibilityMode: boolean,
```
Comment (princemaple) [inline src/lib/datepicker/calendar.ts]: private?

Comment (mmalerba) [inline src/lib/datepicker/calendar.ts]: can't be private since it's used in the template, underscore indicates that its for internal use only

**Comment by princemaple** [comment]
Right! Sorry!

**Pull request merged**